ext/rotoscope/rotoscope.c

```diff
@@ -254,43 +247,15 @@ static Rotoscope *get_config(VALUE self) {
   return config;
 }
 
-void copy_blacklist(Rotoscope *config, VALUE blacklist) {
-  Check_Type(blacklist, T_ARRAY);
-
-  size_t blacklist_malloc_size =
-      RARRAY_LEN(blacklist) * sizeof(*config->blacklist);
-
-  for (long i = 0; i < RARRAY_LEN(blacklist); i++) {
-    VALUE ruby_string = RARRAY_AREF(blacklist, i);
-    Check_Type(ruby_string, T_STRING);
-    blacklist_malloc_size += RSTRING_LEN(ruby_string) + 1;
-  }
-
-  config->blacklist = ruby_xmalloc(blacklist_malloc_size);
-  config->blacklist_size = RARRAY_LEN(blacklist);
-  char *str = (char *)(config->blacklist + config->blacklist_size);
-
-  for (unsigned long i = 0; i < config->blacklist_size; i++) {
-    VALUE ruby_string = RARRAY_AREF(blacklist, i);
-
-    config->blacklist[i] = str;
-    memcpy(str, RSTRING_PTR(ruby_string), RSTRING_LEN(ruby_string));
-    str += RSTRING_LEN(ruby_string);
-    *str = '\0';
-    str++;
-  }
-}
-
 VALUE initialize(int argc, VALUE *argv, VALUE self) {
   Rotoscope *config = get_config(self);
   VALUE output, blacklist;
 
   rb_scan_args(argc, argv, "11", &output, &blacklist);
 
-  if (!NIL_P(blacklist)) {
-    copy_blacklist(config, blacklist);
-  }
+  Check_Type(blacklist, RUBY_T_REGEXP);
```

Comment:
Oh, does the C API allows for type checking? What is the error if blacklist is not a regexp ?

Reply:
Type checking is common in C extensions because they shouldn't crash if used incorrectly form ruby code. Check_Type will raise a TypeError exception if it isn't a Regexp.